src/webapi/date.rs

```diff
@@ -1,25 +1,903 @@
-use std::marker::PhantomData;
-use webcore::try_from::TryInto;
+use std::collections::HashMap;
+use webcore::try_from::{TryFrom, TryInto};
+use webcore::value::{Value, Reference};
 
-/// [(JavaScript docs)](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Date)
-#[derive(Debug)]
-pub struct Date {
-    dummy: PhantomData< () >
-}
-
-impl Date {
-    /// The Date.now() method returns the number of milliseconds elapsed since 1 January 1970 00:00:00 UTC.
+/// A date object
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date)
+pub trait IDate: AsRef< Reference > + TryFrom< Value >  {
+    /// Returns the number of milliseconds elapsed since 1 January 1970 00:00:00 UTC.
     ///
     /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now)
-    pub fn now() -> f64 {
+    #[inline]
+    fn now() -> f64 {
+        js!( return Date.now(); ).try_into().unwrap()
+    }
+
+    /// Accepts the same parameters as the longest form of the constructor,
+    /// and returns the number of milliseconds in a Date object since January 1, 1970, 00:00:00,
+    /// universal time
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/utc)
+    #[inline]
+    fn utc( year: i32, month: u8, day: u8, hour: u8, minutes: u16, seconds: u16, milliseconds: u32) -> i64 {
```

Comment:
1. These arguments should probably all be `i32`.
   * To be consistent with what we return from other methods (e.g. the result of `get_year` should be directly pluggable into `utc` without casting)
   * Putting negative values in here, while kind-of silly, seems to be not disallowed by the [specs](https://tc39.github.io/ecma262/#sec-date.utc), and looks like it's working in both Firefox and Chromium.

2. And the return value should be consistent with `Date::now`, since it's returning the same thing. (So I guess we can make it as `f64` here?)